apps/obsidian/src/components/canvas/DiscourseNodePanel.tsx

```diff
@@ -0,0 +1,231 @@
+            openCreateDiscourseNodeAt({
+              plugin,
+              canvasFile,
+              tldrawEditor: editor,
+              position: pagePoint,
+              initialNodeType: nodeType,
+            });
+          }
+          dragState.set({ name: "idle" });
+          // keep didDrag true through click phase; reset on next tick
+          setTimeout(() => {
+            didDragRef.current = false;
+          }, 0);
+          break;
+        }
+      }
+
+      removeEventListeners();
+    };
+
+    const onPointerDown = (e: React.PointerEvent) => {
+      e.preventDefault();
+      target = e.currentTarget as HTMLButtonElement;
+      target.setPointerCapture(e.pointerId);
+      const nodeTypeId = target.dataset.dg_node_type_id!;
+      if (!nodeTypeId) return;
+      const startPosition = new Vec(e.clientX, e.clientY);
+      dragState.set({ name: "pointing_item", nodeTypeId, startPosition });
+      target.addEventListener("pointermove", handlePointerMove);
+      document.addEventListener("keydown", handleKeyDown);
+    };
+
+    return { handlePointerDown: onPointerDown, handlePointerUp: onPointerUp };
+  }, [dragState, editor, plugin, canvasFile]);
+
+  const state = useValue("dgPanelDragState", () => dragState.get(), [
+    dragState,
+  ]);
+
+  useQuickReactor(
+    "dg-panel-drag-image-style",
+    () => {
+      const current = dragState.get();
+      const imageRef = rDraggingImage.current;
+      const panelRef = rPanelContainer.current;
+      if (!imageRef || !panelRef) return;
+
+      switch (current.name) {
+        case "idle":
+        case "pointing_item": {
+          imageRef.style.display = "none";
+          break;
+        }
+        case "dragging": {
+          const panelRect = panelRef.getBoundingClientRect();
+          const box = new Box(
+            panelRect.x,
+            panelRect.y,
+            panelRect.width,
+            panelRect.height,
+          );
+          const viewportScreenBounds = editor.getViewportScreenBounds();
+          const isInside = Box.ContainsPoint(box, current.currentPosition);
+          if (isInside) {
+            imageRef.style.display = "none";
+          } else {
+            imageRef.style.display = "block";
+            imageRef.style.position = "fixed";
+            imageRef.style.pointerEvents = "none";
+            imageRef.style.left = "0px";
+            imageRef.style.top = "0px";
+            imageRef.style.transform = `translate(${current.currentPosition.x - viewportScreenBounds.x - 25}px, ${current.currentPosition.y - viewportScreenBounds.y - 25}px)`;
+            imageRef.style.width = "50px";
+            imageRef.style.height = "50px";
+            imageRef.style.display = "flex";
+            imageRef.style.alignItems = "center";
+          }
+        }
+      }
+    },
+    [dragState, editor],
+  );
+
+  const nodeTypes = plugin.settings.nodeTypes;
+
+  return (
+    <div className="tlui-style-panel__wrapper p-2" ref={rPanelContainer}>
```

Comment:
Let's go ahead and match the styling in the Roam plugin.
- Color first, so it is aligned
- Color is larger (similar size to other tldraw tools)
- width and style of panel is the same as a shapes style panel ("tlui-layout__top__right etc etc.)

![image.png](https://app.graphite.dev/user-attachments/assets/38208c33-22e6-4c0b-abdd-8b36bb648099.png)